File: .taira-source-c80bf4d85a97/java/iroha_android/src/main/java/org/hyperledger/iroha/android/client/MultisigResponse.java
package org.hyperledger.iroha.android.client;

/** Response emitted by Torii multisig participation endpoints. */
public final class MultisigResponse {
  private final boolean ok;
  private final String resolvedMultisigAccountId;
  private final Boolean submitted;
  private final String proposalId;
  private final String instructionsHash;
  private final String txHashHex;
  private final String executedTxHashHex;
  private final Long creationTimeMs;
  private final String signingMessageB64;

  public MultisigResponse(
      final boolean ok,
      final String resolvedMultisigAccountId,
      final Boolean submitted,
      final String proposalId,
      final String instructionsHash,
      final String txHashHex,
      final String executedTxHashHex,
      final Long creationTimeMs,
      final String signingMessageB64) {
    this.ok = ok;
    this.resolvedMultisigAccountId = resolvedMultisigAccountId;
    this.submitted = submitted;
    this.proposalId = proposalId;
    this.instructionsHash = instructionsHash;
    this.txHashHex = txHashHex;
    this.executedTxHashHex = executedTxHashHex;
    this.creationTimeMs = creationTimeMs;
    this.signingMessageB64 = signingMessageB64;
  }

  public boolean ok() { return ok; }
  public String resolvedMultisigAccountId() { return resolvedMultisigAccountId; }
  public Boolean submitted() { return submitted; }
  public String proposalId() { return proposalId; }
  public String instructionsHash() { return instructionsHash; }
  public String txHashHex() { return txHashHex; }
  public String executedTxHashHex() { return executedTxHashHex; }
  public Long creationTimeMs() { return creationTimeMs; }
  public String signingMessageB64() { return signingMessageB64; }
}

File: kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/client/MultisigResponse.kt
package org.hyperledger.iroha.sdk.client

import org.hyperledger.iroha.sdk.core.model.FeePaymentIntent

/** Response payload returned by Torii multisig participation endpoints. */
data class MultisigResponse(
    val ok: Boolean,
    val resolvedMultisigAccountId: String,
    val submitted: Boolean,
    val proposalId: String?,
    val instructionsHash: String?,
    val txHashHex: String?,
    val executedTxHashHex: String?,
    val creationTimeMs: Long?,
    val feePayment: FeePaymentIntent,
    val transactionPayloadB64: String?,
    val signingMessageB64: String?,
)
